### PR TITLE
feat(charts): Helm chart for the chorus-backend

### PR DESCRIPTION
**Pull request by greut**
This needs to be done after the chart name has been fixed though.

- [x] https://github.com/CHORUS-TRE/chorus-backend/pull/6

DSDIP-426

### CODE REVIEW

File: charts/argo-ci/templates/build-chorus-backend-sensor.yaml [hunk 1]
```diff
@@ -109,6 +128,67 @@ spec:
...
+                        - name: REGISTRY
+                          value: {{ .Values.sensor.registry | quote }}
+                      command: [sh, -e, -c]
+                      # FIXME: the name of the chart is still a little too generic
+                      args:
+                        - >-
+                          helm package deploy/chart;
+                          echo -n "{{`{{inputs.parameters.oci-password}}`}}" | tee /tmp/password
+                          cat /tmp/password | base64 -d | helm registry login -u "{{`{{inputs.parameters.oci-username}}`}}" --password-stdin  "${REGISTRY}";
+                          helm push chart-*.tgz "oci://${REGISTRY}/charts";
```
Comment (greut) [inline charts/argo-ci/templates/build-chorus-backend-sensor.yaml:189]:
```suggestion
                      args:
                        - >-
                          helm package deploy/backend;
                          echo -n "{{`{{inputs.parameters.oci-password}}`}}" | tee /tmp/password
                          cat /tmp/password | base64 -d | helm registry login -u "{{`{{inputs.parameters.oci-username}}`}}" --password-stdin  "${REGISTRY}";
                          helm push backend-*.tgz "oci://${REGISTRY}/charts";
```

Comment (greut) [reply charts/argo-ci/templates/build-chorus-backend-sensor.yaml:189]: @sami-sweng juste?

File: charts/argo-ci/templates/build-chorus-backend-sensor.yaml [hunk 2]
```diff
@@ -17,7 +17,9 @@ spec:
           - path: "body.ref"
             type: string
             value:
+              # The goal is to publish tagged releases.
               - "refs/heads/main"
```
Comment (greut) [inline charts/argo-ci/templates/build-chorus-backend-sensor.yaml:21]:
```suggestion
              - "refs/heads/master"
```

**Comment by greut** [comment]
ping @sami-sweng

**Review by iDmple — APPROVED** [review]

**Pull request merged**